include/itkVkForward1DFFTImageFilter.h

```diff
@@ -45,16 +46,16 @@ namespace itk
  * \sa VkGlobalConfiguration
  * \sa Forward1DFFTImageFilter
  */
-template <typename TInputImage>
+template <typename TInputImage,
+          typename TOutputImage = Image<std::complex<typename TInputImage::PixelType>, TInputImage::ImageDimension>>
 class VkForward1DFFTImageFilter
-  : public Forward1DFFTImageFilter<TInputImage,
-                                 Image<std::complex<typename TInputImage::PixelType>, TInputImage::ImageDimension>>
+  : public Forward1DFFTImageFilter<TInputImage, TOutputImage>
 {
 public:
   ITK_DISALLOW_COPY_AND_MOVE(VkForward1DFFTImageFilter);
 
   using InputImageType = TInputImage;
-  using OutputImageType = Image<std::complex<typename TInputImage::PixelType>, TInputImage::ImageDimension>;
+  using OutputImageType = TOutputImage;
```

Comment:
Because we are no longer calculating `OutputImageType` from `TInputImage`, around here might be a good place to make sure that the input and output image types are compatible in the `FFTImageFilterTraits` sense.  We could do that here if everything needed is already defined by this point.  Similarly for other places where we are no longer setting `OutputImageType` from the `TInputImage`, including the simple cases where we used to write simply `using OutputImageType = TInputImage`

Reply:
Agreed. Note that I tried using `FFTImageFilterTraits` directly for consistency but it became very long and unreadable, so I stuck with `float`, `std::complex<float>`, etc.

Reply:
Sometimes using intermediate `using` statements can neaten things up... e.g., maybe with a `using OutputPixelType = `, a subsequent `using OutputImageType =` won't be so unwieldy.

Reply:
Thanks, will keep in mind for the future. For now I think it's sufficiently readable to keep as-is.